### PR TITLE
kill leaked chrome processes

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Ensure that any leaked chrome process are killed in a finally block.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
This isn't implemented for windows yet, I will wire it up when i get home once I'm confident that it works

File: dev/devicelab/lib/tasks/web_dev_mode_tests.dart [hunk 1]
```diff
@@ -20,122 +20,151 @@ TaskFunction createWebDevModeTest() {
...
+            await process.exitCode;
+          }
+        });
       });
-    });
+    } finally {
+      final Set<String> afterChromeProcesses = await _collectChromeProccesses();
+      final Set<String> newProcesses = afterChromeProcesses.difference(beforeChromeProcesses);
+      for (String processId in newProcesses) {
+        await Process.run('kill', <String>[processId, '-9']);
```
Comment (dnfield) [inline dev/devicelab/lib/tasks/web_dev_mode_tests.dart]: Doesn't this have to be `<String>['-9', processId]`?

File: dev/devicelab/lib/tasks/web_dev_mode_tests.dart [hunk 2]
```diff
@@ -20,122 +20,151 @@ TaskFunction createWebDevModeTest() {
...
       return TaskResult.failure(null);
     }
     return TaskResult.success(null);
   };
 }
+
+/// Collect chrome processes so the test doesn't leak if it exists prematurely.
+Future<Set<String>> _collectChromeProccesses() async {
+  final Set<String> result = <String>{};
+  if (Platform.isMacOS || Platform.isLinux) {
```
Comment (dnfield) [inline dev/devicelab/lib/tasks/web_dev_mode_tests.dart]: Can you reuse `dev/devicelab/lib/framework/running_processes.dart`?

**Comment by jonahwilliams** [comment]
Updated to use the process code you already added yay! Made a minor change to keep pid as a String to avoid the parse/toString roundtrip

**Review by dnfield — APPROVED** [review]
LGTM, I'll close my other one

**Comment by dnfield** [comment]
Only thing is we may still have to log onto the bots and clean up any remaining processes after this lands

**Comment by jonahwilliams** [comment]
Since I don't expect this to effect any other targets (and we need it to get the devicelab back up) I'll land it now then go clean up chrome processes

**Pull request merged**